app/src/main/java/org/oppia/android/app/home/promotedlist/PromotedStoryListView.kt

```diff
@@ -47,12 +48,20 @@ class PromotedStoryListView @JvmOverloads constructor(
    *
    * @param newDataList the new list of stories to present
    */
-  fun setDataList(newDataList: List<PromotedStoryViewModel>) {
-    // Update the adapter and the story list only if the list is new. The parent presenter should
-    // not render promoted stories if the list is empty, but default to showing the last list.
-    if (newDataList != null && !newDataList.isEmpty() && newDataList != dataList) {
-      dataList = newDataList
+  fun setPromotedStoryList(@Nullable newDataList: List<PromotedStoryViewModel>) {
+    // To reliably bind data only after the adapter is created, we manually set the data so we can first
+    // check for the adapter; when using an existing [RecyclerViewBindingAdapter] there is no reliable
+    // way to check that the adapter is created.
+    // This ensures that the adapter will only be created once and correctly rebinds the data.
+    // For more context: https://github.com/oppia/oppia-android/pull/2246#pullrequestreview-565964462
+    if (adapter == null) {
       adapter = createAdapter()
+    }
+
+    if (newDataList == null || newDataList.isEmpty()) {
+      Log.w(PROMOTED_STORY_LIST_VIEW_TAG, ": failed to resolve new story list data")
```

Comment:
```suggestion
      Log.w(PROMOTED_STORY_LIST_VIEW_TAG, "Failed to resolve new story list data")
```

The ':' isn't needed since it'll be included in the logcat output already.

Reply:
Done